src/coord/src/transform_ast.rs

```diff
@@ -0,0 +1,356 @@
+        // we cannot unambiguously resolve.
+        if ident == self.name {
+            self.err = Some(ambiguous_err(self.name, "alias or column"));
+        }
+    }
+    fn visit_object_name(&mut self, object_name: &'ast ObjectName) {
+        self.aggregate_names(&object_name.0);
+    }
+}
+
+struct CreateSqlRewriter {
+    from: Vec<Ident>,
+    to: Vec<Ident>,
+}
+
+impl CreateSqlRewriter {
+    fn rewrite_query_with_qual_depth(
+        from_name: &FullName,
+        to_name: &FullName,
+        qual_depth: usize,
+        query: &mut Query,
+    ) {
+        let (from, to) = match qual_depth {
+            1 => (
+                vec![Ident::new(from_name.item.clone())],
+                vec![Ident::new(to_name.item.clone())],
+            ),
+            2 => (
+                vec![
+                    Ident::new(from_name.schema.clone()),
+                    Ident::new(from_name.item.clone()),
+                ],
+                vec![
+                    Ident::new(to_name.schema.clone()),
+                    Ident::new(to_name.item.clone()),
+                ],
+            ),
+            3 => (
+                vec![
+                    Ident::new(from_name.database.to_string()),
+                    Ident::new(from_name.schema.clone()),
+                    Ident::new(from_name.item.clone()),
+                ],
+                vec![
+                    Ident::new(to_name.database.to_string()),
+                    Ident::new(to_name.schema.clone()),
+                    Ident::new(to_name.item.clone()),
+                ],
+            ),
+            _ => unreachable!(),
+        };
+        let mut v = CreateSqlRewriter { from, to };
+        v.visit_query_mut(query);
+    }
+
+    fn maybe_rewrite_idents(&mut self, h: &mut Vec<Ident>) {
+        // We don't want to rewrite if the item we're rewriting is shorter than
+        // the values we want to replace them with.
+        if h.len() < self.from.len() {
+            return;
+        }
+        let n = &self.from;
+        for i in 0..h.len() - n.len() + 1 {
+            // If subset of `h` matches `self.from`...
+            if h[i..i + n.len()] == n[..] {
+                // ...splice `self.to` into `h` in that subset's location.
+                h.splice(i..i + n.len(), self.to.iter().cloned());
+                return;
+            }
+        }
+    }
+}
+
+impl<'ast> VisitMut<'ast> for CreateSqlRewriter {
+    fn visit_query_mut(&mut self, query: &'ast mut Query) {
+        visit_mut::visit_query_mut(self, query);
+    }
+    fn visit_expr_mut(&mut self, e: &'ast mut Expr) {
+        match e {
+            Expr::Identifier(ref mut i) | Expr::QualifiedWildcard(ref mut i) => {
+                self.maybe_rewrite_idents(i);
+            }
+            _ => visit_mut::visit_expr_mut(self, e),
+        }
+    }
+    fn visit_object_name_mut(&mut self, object_name: &'ast mut ObjectName) {
+        self.maybe_rewrite_idents(&mut object_name.0);
+    }
+}
```

Comment:
Could you chuck this whole module into the `sql` crate somewhere? I don't have a particular opinion on where, (besides that the coord crate shouldn't be in the business of reasoning about SQL idents and such).